### PR TITLE
Sanity test fixups: AWS VPC related modules

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY
- remove sanity ignore entries for AWS VPC related modules
- remove defaults from required options
- Avoid blocklisted variable name '_'
- Add boilerplate
- Remove explicit definition of 'region' from argument spec it's automatically added by ec2_argument_spec and redefining removes the aliases

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/amazon/ec2_vpc_dhcp_option.py
lib/ansible/modules/cloud/amazon/ec2_vpc_dhcp_option_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_egress_igw.py
lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint.py
lib/ansible/modules/cloud/amazon/ec2_vpc_endpoint_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_igw.py
lib/ansible/modules/cloud/amazon/ec2_vpc_igw_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway.py
lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_net.py
lib/ansible/modules/cloud/amazon/ec2_vpc_net_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_peer.py
lib/ansible/modules/cloud/amazon/ec2_vpc_peering_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_route_table.py
lib/ansible/modules/cloud/amazon/ec2_vpc_route_table_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py
lib/ansible/modules/cloud/amazon/ec2_vpc_subnet_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_vgw.py
lib/ansible/modules/cloud/amazon/ec2_vpc_vgw_info.py
lib/ansible/modules/cloud/amazon/ec2_vpc_vpn.py
lib/ansible/modules/cloud/amazon/ec2_vpc_vpn_info.py
test/sanity/ignore.txt
test/units/modules/cloud/amazon/test_ec2_vpc_nat_gateway.py
test/units/modules/cloud/amazon/test_ec2_vpc_vpn.py

### CODE REVIEW

**Review by gundalow — APPROVED** [review]
Great work, than you.

**Review by gundalow — CHANGES_REQUESTED** [review]
Unit test failures, I haven't looked into why they are failing https://app.shippable.com/github/ansible/ansible/runs/148601/7/tests

**Review by felixfontein — COMMENTED** [review]
Looks good to me. You could also use `dummy` instead of `_`, instead of removing it completely.

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway.py [hunk 1]
```diff
@@ -53,22 +58,24 @@
       - Deallocate the EIP from the VPC.
       - Option is only valid with the absent state.
       - You should use this with the wait option. Since you can not release an address while a delete operation is happening.
-    default: 'yes'
+    default: false
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway.py:61]: Would you mind updating the examples to use true/false as well?

File: lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway_info.py [hunk 2]
```diff
@@ -23,11 +23,14 @@
   nat_gateway_ids:
     description:
       - Get details of specific nat gateway IDs
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway_info.py]:
```suggestion
      - List of specific nat gateway IDs to get details for.
```

File: lib/ansible/modules/cloud/amazon/ec2_vpc_net.py [hunk 3]
```diff
@@ -136,7 +143,7 @@
     classic_link_enabled:
       description: indicates whether ClassicLink is enabled
       returned: always
-      type: NoneType
+      type: bool
       sample: null
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_net.py]:
```suggestion
       sample: false
```

File: lib/ansible/modules/cloud/amazon/ec2_vpc_peering_info.py [hunk 4]
```diff
@@ -22,11 +22,14 @@
   peer_connection_ids:
     description:
       - Get details of specific vpc peer IDs
```
Comment (jillr) [inline lib/ansible/modules/cloud/amazon/ec2_vpc_peering_info.py]:
```suggestion
      - List of specific vpc peer IDs to get details for.
```

**Comment by jillr** [comment]
Also, on this one and 64064 you'll need to add test/sanity/ignore.txt entries for the `option-incorrect-version-added` failures on the module docs that are backfilling that data, then we can remove them in a future PR.

**Review by jillr — APPROVED** [review]
LGTM once CI passes.

ETA: As we discussed on irc, force merging to save having to do multiple sanity/ignores.txt PRs just to make the tests happy.

**Comment by mattclay** [comment]
Do not add ignores for `option-incorrect-version-added`. Doing so will cause test failures after merge. Instead, once all the tests are passing (except for the `option-incorrect-version-added` failure), the PR can be manually merged. Just make sure to check over the entire Shippable results to make sure that is the only remaining error before merging.

**Review by felixfontein — APPROVED** [review]

**Pull request merged**